presto-main/src/main/java/io/prestosql/Session.java

```diff
@@ -293,6 +293,31 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
         requireNonNull(transactionManager, "transactionManager is null");
         requireNonNull(accessControl, "accessControl is null");
 
+        Session sessionWithTransaction = new Session(
```

Comment:
return it also instead of creating new session object

Reply:
revert